### PR TITLE
Gutenboarding: Add `/new/plans` route that shows plans grid

### PR DESCRIPTION
**Pull request by p-jackson**
#### Changes proposed in this Pull Request

Adds plans step at `/new/plans`, removing the need for the `prelaunch` frankenflow.

Combined with #42026, this means users who choose the eCommerce plan will have their site transfered to atomic and will go through the store onboarding.

* Plan Step comes after font step if:
  * the user chooses a paid domain
  <s>* and the haven't already choosen a domain
    * either by using the Plans Modal button in the top right
    * or by specifying one as a route param</s> https://github.com/Automattic/wp-calypso/pull/42028#pullrequestreview-408976078
* Switching between plans immediately updates the store (can be seen by the plan label in the top-right).
  * This is <s>different from</s> now also the Plans Modal behaviour https://github.com/Automattic/wp-calypso/pull/42028#pullrequestreview-408976078
* User can go back to the Font Selection step
* The plans grid auto-selects the plan specified in the route
* The Plans Step fires tracks events
  * `calypso_newsite_step_enter`
  * `calypso_newsite_step_leave`, which has a `selected_plan` property
* Remove link to `prelaunch` flow

#### Testing instructions

* Testing requires the `gutenboarding/plans-grid` feature flag, so should be tested locally.
  * `/new`
  * Choose a paid domain
  * The next step after choosing a font should be the Plans Step
  * If there was no previous interaction with Plans Button, _Premium Plan_ should be pre-selected
  * Site should be created as expected:
    * If free plan then go straight to block editor
    * If paid plan then go to checkout, then block editor when complete
    * If eCommerce plan then go to checkout, then on to the store onboarding
* Also test on `calypso.live` to check gutenboarding isn't broken when the feature flag is disabled.

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/plans/index.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+					>
+						{ __( 'Continue' ) }
+					</Button>
+				) }
+				onPlanChange={ setPlan }
+				selectedPlanSlug={ plan.getStoreSlug() }
+				cancelButton={
+					<Link className="plans__back-link" isLink to={ makePath( Step.Style ) }>
+						{ __( 'Go back' ) }
+					</Link>
```
Comment (razvanpapadopol) [inline client/landing/gutenboarding/onboarding-block/plans/index.tsx:72]: Styling of the action buttons is a bit different than the one used for modal.

Plans Modal:
Screenshot 2020-05-11 at 10 13 48
Screenshot 2020-05-11 at 10 13 36

Plans Page:

**Comment by ramonjd** [comment]
What's the advantage of having both the modal and the route looking exactly the same? 

It confused me a little bit, especially since I can open the plans modal when I'm on the plans page :)

May-11-2020 19-44-00

**Comment by razvanpapadopol** [comment]
> I can open the plans modal when I'm on the plans page

Maybe this is more a design question but for now I can just disable the button when you are on the Plans page.

**Comment by p-jackson** [comment]
Thanks for picking this up @razvanpapadopol 
I wasn't entirely happy with how the step wouldn't stay in the flow if you used the back button, so this is an improvement.
It does mean that plans step will appear even if the url has already specified one, but that's minor by comparison. We might want to revisit this when we have buttons landing directly on those routes (I don't think we have any yet).

**Comment by ramonjd** [comment]
Looking great folks!

Two things I noticed, and I wanted to confirm if it was just me:


1. I selected a paid domain and proceeded to the plans step. Here I selected a free plan and was taken straight to the editor. The behaviour is right, but should we think about warning the user, who has actively selected a paid domain, that their domain selection is forfeit when opting for a free account?
2. _After having created an account and a site_, if I return to `/new` and select a paid domain, the Free plan doesn't change: free-plan-forever


Kicked off the e2e tests again - the failures seemed to be unrelated

**Comment by andrewserong** [comment]
Just tested as well, and I can recreate issue 3 that Ramon mentioned: 

> 3. _After having created an account and a site_, if I return to `/new` and select a paid domain, the Free plan doesn't change:

Do we need to add another call to `resetPlan` in these places?:

* "Has paid domain and plans grid isn't enabled" — we possibly don't _need_ it, but it might be cleaner to add it anyway
https://github.com/automattic/wp-calypso/blob/b36d6ed2b04789fcb51e2070d2d082f760d69d3a/client/landing/gutenboarding/hooks/use-on-site-creation.ts#L133

* Just before `resetOnboardStore()` for creating a free site — this appears to be the culprit:
https://github.com/automattic/wp-calypso/blob/b36d6ed2b04789fcb51e2070d2d082f760d69d3a/client/landing/gutenboarding/hooks/use-on-site-creation.ts#L146

**Comment by p-jackson** [comment]
> I selected a free plan and was taken straight to the editor. The behaviour is right, but should we think about warning the user, who has actively selected a paid domain, that their domain selection is forfeit when opting for a free account?

@ramonjd yeah some messaging would be good. This does as bad a job as the current franken-prelaunch-flow so I think we should fix it seperately.

It'd be nice to have something we could show inline. What if the "free" column in the plans grid had an extra entry like `❌ no custom domains` when a paid domain is selected? @dubielzyk 

> _After having created an account and a site_, if I return to `/new` and select a paid domain, the Free plan doesn't change

Thanks for the tips @andrewserong 

> for some reason four new sites are created instead of one

I ... had not seen that 😓

File: client/landing/gutenboarding/components/header/index.tsx [hunk 2]
```diff
@@ -129,13 +128,9 @@ const Header: React.FunctionComponent = () => {
 
 	React.useEffect( () => {
 		if ( newUser && newUser.bearerToken && newUser.username && ! newSite ) {
```
Comment (andrewserong) [inline client/landing/gutenboarding/components/header/index.tsx:130]: Would it help to add a `! isCreatingSite` check to this `if` statement, like we're using in the `OnboardingEdit` component? Ref: https://github.com/automattic/wp-calypso/blob/fbd2194d6426b84f607a67ffd5e6bc9646820d35/client/landing/gutenboarding/onboarding-block/edit.tsx#L38 and https://github.com/automattic/wp-calypso/blob/fbd2194d6426b84f607a67ffd5e6bc9646820d35/client/landing/gutenboarding/onboarding-block/edit.tsx#L85

Comment (p-jackson) [reply client/landing/gutenboarding/components/header/index.tsx:130]: I managed to track down what was re-running the side-effect 607c9c6

Curse these `useEffect()`s :shake-fist:

Comment (andrewserong) [reply client/landing/gutenboarding/components/header/index.tsx:130]: Ah, gotcha. Nice one!
Yes, it's the one thing that really frustrates me about using hooks over class component lifecycle methods, it always seemed like it was going to be simpler to reason about, but then we have situations like this one! 😄

**Comment by p-jackson** [comment]
Ok, I think that's #2 and #3 fixed. One more time!

**Comment by andrewserong** [comment]
I'll give this another look in a little bit. Just adding that I believe your fix here will fix this issue: https://github.com/Automattic/wp-calypso/issues/42086

**Pull request merged**